AutoRaise.mm

```diff
@@ -475,48 +599,30 @@ CGEventRef eventTapHandler(CGEventTapProxy proxy, CGEventType type, CGEventRef e
 }
 
 #define POLLING_MS 20
+#define VERSION "1.8"
 int main(int argc, const char * argv[]) {
     @autoreleasepool {
-        if (argc >= 3) {
-            NSUserDefaults * standardDefaults = [NSUserDefaults standardUserDefaults];
-            delayCount = abs((int) [standardDefaults integerForKey: @"delay"]);
-            warpMouse = argc != 3 ;
-            if (argc >= 7) {
-                warpX = [standardDefaults floatForKey: @"warpX"];
-                warpY = [standardDefaults floatForKey: @"warpY"];
-                if (argc >= 8) {
-                    cursorScale = [standardDefaults floatForKey: @"scale"];
-                }
-            }
-        } else {
-            NSString * home = NSHomeDirectory();
-            NSFileHandle * delayFile = [NSFileHandle fileHandleForReadingAtPath:
-                [NSString stringWithFormat: @"%@/AutoRaise.delay", home]];
-            if (delayFile) {
-                delayCount = abs([[[NSString alloc] initWithData:
-                    [delayFile readDataOfLength: 2] encoding:
-                    NSUTF8StringEncoding] intValue]);
-                [delayFile closeFile];
-            }
-            NSFileHandle * warpFile = [NSFileHandle fileHandleForReadingAtPath:
-                [NSString stringWithFormat: @"%@/AutoRaise.warp", home]];
-            if (warpFile) {
-                warpMouse = true;
-                NSString * line = [[NSString alloc] initWithData:
-                    [warpFile readDataOfLength:11] encoding:
-                    NSUTF8StringEncoding];
-                NSArray * components = [line componentsSeparatedByString: @" "];
-                if (components.count >= 1) { warpX = [[components objectAtIndex:0] floatValue]; }
-                if (components.count >= 2) { warpY = [[components objectAtIndex:1] floatValue]; }
-                if (components.count >= 3) { cursorScale = [[components objectAtIndex:2] floatValue]; }
-                [warpFile closeFile];
+        printf("\nv%s by sbmpost(c) 2021, usage:\nAutoRaise -delay <1=%dms> [-warpX <0.5> -warpY <0.5> -scale <2.0>]", VERSION, POLLING_MS);
+        
+        ConfigClass * config = [[ConfigClass alloc]init];
+        
+        if (argc > 1) {                             // if there are any arguments, we would use only them
+            [config readCommandLineConfig];
+        } else {                                    // otherwise try to read
+            if (not [config readOriginalConfig]) {  // original config files
```

Comment:
This logic probably disappears if you decide to merge NSDictionaries.

Reply:
But we need to check NSDictionary values every time to know - is there needs to go to next configuration method or not.
Isn't it?

Reply:
Yep! 
But it's return NSString and we need to make BOOL for conditions.
And I found that in Objective-C for any non-null string, except 1 YES and true,  return false for [myString boolValue] :/

Reply:
That is one way of doing it. The other is to blindly overwrite the dictionary keys in priority order, something like:


D = FromCmdLine(
      FromOld(
        FromNew(
          Defaults)));

Each method in this chain would accept a dictionary and return a new one, overwriting a value if the key for it is present

Reply:
Hmm..
But I wanted to do another logic:
if there is *any* arguments from command-line => we should to ignore all config files.

Reply:
The behaviour would indeed be different. And maybe a bit strange for the end user who suddenly is able to provide warpX via the commandline and warpX through a config file. This can however also be done while chaining methods by saying that if there is already one single key present in the previous dictionary, then return that dictionary instead of the new one. Again now absolute right or wrong here ;-)

Reply:
Ok!
I understand that.

Reply:
It is acceptable :-) I might experiment with it myself at some point however. But let us for now focus on getting your PR merged ;-). I think if you double check the other comments and fix what you think needs fixing, then I will test it thoroughly and we merge.

Reply:
Haha, cool :) I am signing out for today, take it easy and thanks for the effort!